src/app/components/table/table.ts

```diff
@@ -5187,7 +5187,7 @@ export class ReorderableRow implements AfterViewInit {
                             <ng-template *ngTemplateOutlet="addRuleIconTemplate"></ng-template>
                         </button>
                     </div>
-                    <div class="p-column-filter-buttonbar">
+                    <div class="p-column-filter-buttonbar" *ngIf="showButtons">
```

Comment:
Is this the only change needed to fix issue #15937? I'm assuming this is not needed for issue #15937

Note: i did not try to verify the change